nautobot/core/templatetags/buttons.py

```diff
@@ -124,6 +126,140 @@ def job_import_url(content_type):
+        bulk_delete_url (str): The URL for the bulk delete action.
+        permissions (dict): A dictionary of specific permissions for the actions.
+    """
+
+    bulk_action_buttons = []
+
+    render_edit_button = bool(bulk_edit_url and permissions["change"])
+    render_static_group_assign_button = bool(
+        content_type.model_class().is_dynamic_group_associable_model and perms["extras"]["add_staticgroupassociation"]
+    )
+    render_delete_button = bool(bulk_delete_url and permissions["delete"])
+    bulk_action_button_count = sum([render_edit_button, render_static_group_assign_button, render_delete_button])
+
+    if bulk_action_button_count == 0:
+        return {
+            "bulk_action_buttons": bulk_action_buttons,
+        }
+
+    params = ("?" + request.GET.urlencode()) if request.GET else ""
+
+    primary_button_fragment = child_button_fragment = """
+        <button {attrs}>
+            <span class="{icon}" aria-hidden="true"></span> {label}
+        </button>
+    """
+
+    edit_button_classes = "btn btn-sm btn-warning"
+    delete_button_classes = "btn btn-sm btn-danger"
+    static_group_button_classes = "btn btn-sm btn-primary"
+    static_group_icon = "mdi mdi-group"
+
+    if bulk_action_button_count > 1:
+        child_button_fragment = f"<li>{primary_button_fragment}</li>"
+        delete_button_classes = "text-danger"
+        static_group_button_classes = "text"
+        static_group_icon += " text-muted"
+
+    if render_edit_button:
+        bulk_action_buttons.append(
+            format_html(
+                primary_button_fragment,
+                label="Edit Selected",
+                attrs=render_tag_attrs(
+                    {
+                        "class": edit_button_classes,
+                        "formaction": reverse(bulk_edit_url) + params,
+                        "type": "submit",
+                    }
+                ),
+                icon="mdi mdi-pencil",
+            ),
+        )
+        if bulk_action_button_count > 1:
+            bulk_action_buttons[0] += mark_safe(  # noqa: S308 -- suspicious-mark-safe-usage
+                f"""
+                <button type="button" data-toggle="dropdown" class="{edit_button_classes} dropdown-toggle" aria-haspopup="true">
+                    <span class="caret"></span>
+                    <span class="sr-only">Toggle Dropdown</span>
+                </button>
+                """
+            )
+
+    # Render a generic "Bulk Actions" dropup button if the edit button is not present
+    elif bulk_action_button_count > 1:
+        bulk_action_buttons.append(
+            mark_safe(  # noqa: S308 -- suspicious-mark-safe-usage
+                """
+                <button type="button" class="btn btn-sm btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true">
+                    <span class="mdi mdi-plus-thick" aria-hidden="true"></span> Bulk Actions <span class="caret"></span>
```

Comment:
Feels off to me to use `mdi-plus-thick` for the generic "Bulk Actions" button. Can we omit an icon entirely in this case?